connector_prestashop/models/product_template/importer.py

```diff
@@ -0,0 +1,708 @@
+            self.backend_record,
+            record_id,
+            record=record,
+            **kwargs
+        )
+
+
+class ProductInventoryImporter(Component):
+    _name = 'prestashop._import_stock_available.importer'
+    _inherit = 'prestashop.importer'
+    _apply_on = '_import_stock_available'
+
+    def _get_quantity(self, record):
+        filters = {
+            'filter[id_product]': record['id_product'],
+            'filter[id_product_attribute]': record['id_product_attribute'],
+            'display': '[quantity]',
+        }
+        quantities = self.backend_adapter.get(filters)
+        all_qty = 0
+        quantities = quantities['stock_availables']['stock_available']
+        if isinstance(quantities, dict):
+            quantities = [quantities]
+        for quantity in quantities:
+            all_qty += int(quantity['quantity'])
+        return all_qty
+
+    def _get_binding(self):
+        record = self.prestashop_record
+        if record['id_product_attribute'] == '0':
+            binder = self.binder_for('prestashop.product.template')
+            return binder.to_internal(record['id_product'])
+        binder = self.binder_for('prestashop.product.combination')
+        return binder.to_internal(record['id_product_attribute'])
+
+    def _import_dependencies(self):
+        """ Import the dependencies for the record"""
+        record = self.prestashop_record
+        self._import_dependency(
+            record['id_product'], 'prestashop.product.template'
+        )
+        if record['id_product_attribute'] != '0':
+            self._import_dependency(
+                record['id_product_attribute'],
+                'prestashop.product.combination'
+            )
+
+    def _check_in_new_connector_env(self):
+        # not needed in this importer
+        return
+
+    def run(self, prestashop_id, record=None, **kwargs):
+        assert record
+        self.prestashop_record = record
+        return super(ProductInventoryImporter, self).run(
+            prestashop_id, **kwargs
+        )
+
+    def _import(self, binding, **kwargs):
+        record = self.prestashop_record
+        qty = self._get_quantity(record)
+        if qty < 0:
+            qty = 0
+        if binding._name == 'prestashop.product.template':
+            products = binding.odoo_id.product_variant_ids
+        else:
+            products = binding.odoo_id
+
+        location = (self.backend_record.stock_location_id or
+                    self.backend_record.warehouse_id.lot_stock_id)
+        for product in products:
+            vals = {
+                'location_id': location.id,
+                'product_id': product.id,
+                'new_quantity': qty,
+            }
+            template_qty = self.env['stock.change.product.qty'].create(vals)
+            template_qty.with_context(
+                active_id=product.id,
+                connector_no_export=True,
+            ).change_product_qty()
```

Comment:
It is not a bug, but I suggest to only call change_product when product's type is 'product'.
This way, it will not try to import stock to the products that is not required to check for the stock ('consu' and 'service' types), if not for each 'consu' and 'service' product an exception is raised.